nemo/core/classes/modelPT.py

```diff
@@ -316,6 +329,144 @@ def val_dataloader(self):
     #     if self._test_dl is not None:
     #         return self._test_dl
 
+    def validation_epoch_end(
+        self, outputs: Union[List[Dict[str, torch.Tensor]], List[List[Dict[str, torch.Tensor]]]]
+    ) -> Dict[str, Dict[str, torch.Tensor]]:
+        """
+        Default DataLoader for Validation set which automatically supports multiple data loaders
+        via `multi_validation_epoch_end`.
+
+        If multi dataset support is not required, override this method entirely in base class.
+        In such a case, there is no need to implement `multi_validation_epoch_end` either.
+
+        Note:
+            If more than one data loader exists, and they all provide `val_loss`,
+            only the `val_loss` of the first data loader will be used by default.
+            This default can be changed by passing the special key `val_loss_idx: int`
+            inside the `validation_ds` config.
```

Comment:
I do not understand this assumption...

Shouldn't the `mean` by the default? Or is it something PTL assumes/enforces?

Reply:
These val_loss come from different data loaders. As in different datasets (potentially) - say dev_clean and dev_other, test_clean and test_other. Average over different datasets is not a correct assumption and there is no literature to support this is a good way to select "model which does best on validation set".

val_loss is not used for computing loss or gradients, only for monitoring the "Best validation loss" in order to save the "best" model weights. This API assumes the 1st (0th index) data loader serves as the basis for this "best model weights" selection, but also offers the overrides `val_loss_idx` and `test_loss_idx` as commandline overrides if we want to choose (say) dev-other instead of dev-clean for "best" model.